Framework/MDAlgorithms/src/Integrate3DEvents.cpp

```diff
@@ -607,12 +614,16 @@ double Integrate3DEvents::numInEllipsoidBkg(
   auto endIndex = eventVec.size();
   if (useOnePercentBackgroundCorrection) {
     // Remove top 1% of background
-    std::sort(eventVec.begin(), eventVec.end());
+    std::sort(eventVec.begin(), eventVec.end(),
+              [](std::pair<double, double> a, std::pair<double, double> b) {
```

Comment:
Good use of lambda. Minor style point: I would expect this to have each parameter as `const std::pair<double, double> &a, const std::pair<double, double> &b` to avoid copying the pair as it sorts and calls the lambda.